inference-engine/thirdparty/clDNN/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -2396,7 +2398,207 @@ INSTANTIATE_TEST_SUITE_P(DISABLED_REORDER,
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        data("weights_sec", get_mem(get_weights_layout(p))),
+        reorder("reorder_fp32", "input", format::bfyx, data_types::f32),
+        convolution("conv_prim", "reorder_fp32", { "weights" }, { "bias" }, 1, p.stride, p.pad, tensor{1}, p.in_shape, data_types::u8, false),
+        reorder("reorder_conv", "conv_prim", reorder_layout),
+        convolution("conv_output", "reorder_conv", { "weights_sec" }, 1, p.stride, p.pad),
+        reorder("reorder_bfyx", "conv_output", format::b_fs_yx_fsv32, data_types::f32),
+        resample("resample", "reorder_bfyx", p.out_shape, 1),
+        reorder("reorder_output", "resample", p.default_format, data_types::f32)
+    );
+
+    execute(p);
+
+    EXPECT_EQ(check_not_removed(p, "reorder_conv"), true);
+}
+
+TEST_P(testing_removal_reorder, removal_metaonly_reorder) {
+    auto p = GetParam();
+    create_topologies(input_layout("input", get_input_layout(p)),
+                reorder("reorder_input", "input", format::b_fs_yx_fsv16, data_types::f16),
+                data("weights", get_mem(get_weights_layout(p))),
+                data("bias1", get_mem(get_bias_layout(p))),
+                reorder("reorder_bias1", "bias1", format::b_fs_yx_fsv16, data_types::f16),
+                convolution("conv_prim", "reorder_input", {"weights"}, std::vector<primitive_id>{}, 1, p.stride, p.pad),
+                eltwise("add_bias1", {"conv_prim", "reorder_bias1"}, eltwise_mode::sum),
+                reorder("reorder_bfyx", "add_bias1", p.default_format, data_types::f16)
+    );
+
+    execute(p);
+
+    EXPECT_EQ(check_removed(p, "reorder_bias1"), true);
+}
+
+TEST_P(testing_removal_reorder, removal_padded_reorder_base) {
+    auto p = GetParam();
+    layout reorder_layout(data_types::f16, format::b_fs_yx_fsv16, p.in_shape, padding({0, }, 0));
+
+    create_topologies(input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        reorder("reorder_fp32", "input", format::bfyx, data_types::f16),
+        convolution("conv_prim", "reorder_fp32", { "weights" }, 1, p.stride, p.pad),
+        reorder("reorder_conv", "conv_prim", reorder_layout),
+        convolution("conv_output", "reorder_conv", { "weights" }, 1, p.stride, p.pad),
+        reorder("reorder_output", "conv_output", p.default_format, data_types::f32)
+    );
+
+    auto build_opts = build_options();
+    build_opts.set_option(build_option::optimize_data(true));
+    implementation_desc impl = { format::b_fs_yx_fsv16, std::string(""), impl_types::ocl };
+    build_opts.set_option(build_option::force_implementations({ {"conv_output", impl} }));
+
+    setup_with_build_ops(build_opts);
+
+    execute(p);
+
+    EXPECT_EQ(check_removed(p, "reorder_conv"), true);
+}
+
+TEST_P(testing_removal_reorder, removal_padded_reorder) {
```

Comment:
Could you add brief description on each tests?

Reply:
Done